### PR TITLE
A few miscellaneous things

### PR DESCRIPTION
**Pull request by mbrandonw**
* I had a few instances of `PrinterParser` that I renamed to `ParserPrinter`.
* Added the `require` partial iso
* Fixed `queryParam` to use a `PartialIso<String?, A>`
* Enabled caching of build artifacts on circle and travis. I've been wanting to play with this for awhile. On circle, caching works very well for `swift test`, but I can't see to get `xcodebuild` to recognize the cache. [This](https://blog.bitrise.io/60-percent-faster-builds-force-xcode-use-caching) article seems to indicate that the timestamps of the files change with each `git clone` on CI, and that messes with `xcodebuild`'s ability to cache, but I tried normalizing the timestamps and still couldn't get a consistent cache. Also on travis caching doesn't help a ton because most of the time is spent preparing the environment.
* Added a test phase for playgrounds. This only works on circle, and it could be improved a bit, but at least it works.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+	set -o pipefail && \
+	xcodebuild test \
+		-scheme ParserPrinter-Package \
+		-destination platform="iOS Simulator,name=iPhone XR,OS=12.0" \
+		-derivedDataPath ./.derivedData \
+		| xcpretty
+
+test-playgrounds:
+	swift \
+		-F .derivedData/Build/Products/Debug/ \
```
Comment (mbrandonw) [inline Makefile]: we can use the build artifacts from `xcodebuild` to link all the necessary libs and then execute the swift playground.

Comment (mbrandonw) [reply Makefile]: also this should be beefed up to compile all of the playgrounds in the repo, but my shell-fu isn't good enough for that. may need some help

Comment (stephencelis) [reply Makefile]: Made it a bit more general. May need to beef up even more for source dependencies...

Comment (mbrandonw) [reply Makefile]: nice thanks! 

yeah I don't think we need to worry about sources. I imagine anything we wanna put in there should prob just go in the library itself.

Comment (stephencelis) [reply Makefile]: We could also try something like this: https://flight.school/articles/running-xcode-playgrounds-on-travis-ci/

Comment (mbrandonw) [reply Makefile]: yeah I looked into that. but also we could concat everything in the sources directory to the `Contents.swift` right? seems simpler

**Pull request merged**